activerecord/test/cases/schema_dumper_test.rb

```diff
@@ -400,6 +400,41 @@ def down
     end
   end
 
+  class UnknownColumnTypeMigration < ActiveRecord::Migration::Current
+    def up
+      create_table("forecasts") do |t|
+        if current_adapter?(:PostgreSQLAdapter)
+          execute "CREATE TYPE conditions AS ENUM ('sunny', 'rainy')"
+        end
+
+        t.column :weather, :conditions
+      end
+    end
+
+    def down
+      drop_table("forecasts")
+      execute "DROP TYPE conditions" if current_adapter?(:PostgreSQLAdapter)
+    end
+  end
+
+  unless current_adapter?(:Mysql2Adapter)
```

Comment:
on a scale of 🌈 to 🌞 this makes me feel a bit 🌦.

Before checking the current adapters, what would happen is that on the `mysql` and `postrgresql` adapters the `t.column :weather, :conditions` would fail with a syntax error since conditions is not a valid data type in the generated `sql`, this happily worked on `sqlite3` however. 

This invalid sql statement would cause the `migration.migrate(:up)` call in the test to fail, and then the `migration.migrate(:down)` in the ensure block would throw an exception since the table was never created. 

I thought we had a couple of approaches: 

1) stick this whole test in `unless memory_db || current_adapter?(:Sqlite3Adapter)?` and just run it for sqlite, or just run it for `postgres`.

2) add some conditionals and a `CREATE TYPE` to trigger the error in Postgres, leave as is for `sqlite` and skip for mysql since it doesn't support user defined types, this issue shouldn't occur in mysql. 

I thought we would get some more coverage across adapters with 2 so I went with two, but it would be more 🌞 if we didn't have this many conditionals 🤷‍♂